scrapi/harvesters/plos.py

```diff
@@ -108,85 +106,21 @@ def copy_to_unicode(self, element):
         else:
             return unicode(element, encoding=DEFAULT_ENCODING)
 
-    def get_ids(self, raw_doc, record):
-        doi = record.xpath('//str[@name="id"]/node()')[0]
-        ids = {
-            'doi': self.copy_to_unicode(doi),
-            'serviceID': raw_doc.get('docID'),
-            'url': 'http://dx.doi.org/{}'.format(doi)
-        }
-        return ids
-
-    def get_contributors(self, record):
-        contributor_list = []
-        contributors = record.xpath('//arr[@name="author_display"]/str/node()') or ['']
-        for person in contributors:
-            name = HumanName(person)
-            contributor = {
-                'prefix': name.title,
-                'given': name.first,
-                'middle': name.middle,
-                'family': name.last,
-                'suffix': name.suffix,
-                'email': '',
-                'ORCID': ''
-            }
-            contributor_list.append(contributor)
-        return contributor_list
-
-    def get_properties(self, record):
-        properties = {
-            'journal': (record.xpath('//str[@name="journal"]/node()') or [''])[0],
-            'eissn': (record.xpath('//str[@name="eissn"]/node()') or [''])[0],
-            'articleType': (record.xpath('//str[@name="article_type"]/node()') or [''])[0],
-            'score': (record.xpath('//float[@name="score"]/node()') or [''])[0],
+    schema = {
+        'id': {
+            'doi': '//str[@name="id"]/node()',
+            'serviceID': '//str[@name="id"]/node()',
+            'url': ('//str[@name="id"]/node()', lambda x: 'http://dx.doi.org/{}'.format(x))
+        },
+        'contributors': ('//arr[@name="author_display"]/str/node()', default_name_parser),
+        'dateUpdated': ('//date[@name="publication_data"]/node()', lambda x: parse(x).isoformat().decode('utf-8')),
+        'tags': ('//str[@name="id"]/node()', lambda x: []),  # TODO Have some kind of skip value
```

Comment:
right now the transformer system doesn't allow skips/noops, which means I needed to stick a dummy value in. Definitely next on my priority list, but a bit out of scope for the PR. Will remove this todo when I add a skip/noop branch.